### ISSUE TITLE
Error loading genbank record using secmet record

### ISSUE THREAD

**Issue by seth-1** [antismash/antismash#233]
Hi guys,
I'm trying to use ```Record.from_genbank``` on this [file](https://antismash.secondarymetabolites.org/upload/bacteria-47148450-7153-4868-9792-fa438d6ad626/Amycolatopsis_keratinaphila.gbk). The run has been generated beginning of February and I just installed antismash again using pip (see below for versions).

Trying to load the file above I get several Exceptions. I used to run the function without problem a year or so ago.

```python
from antismash.common.secmet import Record

 rec = Record.from_genbank("Amycolatopsis_keratinaphila.gbk")

Traceback (most recent call last):
  File "/.../lib/python3.7/site-packages/antismash/common/secmet/record.py", line 340, in get_domain_by_name
    return self._domains_by_name[name]
KeyError: 'nrpspksdomains_ctg1_2108_Condensation_Starter.1'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.../lib/python3.7/site-packages/antismash/common/secmet/features/module.py", line 201, in from_biopython
    domains = [record.get_domain_by_name(domain) for domain in domain_names]
  File "/.../lib/python3.7/site-packages/antismash/common/secmet/features/module.py", line 201, in <listcomp>
    domains = [record.get_domain_by_name(domain) for domain in domain_names]
  File "/.../lib/python3.7/site-packages/antismash/common/secmet/record.py", line 342, in get_domain_by_name
    raise KeyError("record %s contains no domain named %s" % (self.id, name))
KeyError: 'record NZ_LT629789.1 contains no domain named nrpspksdomains_ctg1_2108_Condensation_Starter.1'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.../lib/python3.7/site-packages/antismash/common/secmet/record.py", line 762, in from_biopython
    record.add_feature(cls.from_biopython(feature, record=record))
  File "/.../lib/python3.7/site-packages/antismash/common/secmet/features/module.py", line 204, in from_biopython
    bio_feature.location, err))
ValueError: record does not contain domain referenced by module at [2331621:2334660](+): 'record NZ_LT629789.1 contains no domain named nrpspksdomains_ctg1_2108_Condensation_Starter.1'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.../lib/python3.7/site-packages/antismash/common/secmet/record.py", line 774, in from_genbank
    records.append(Record.from_biopython(bio, taxon))
  File "/.../lib/python3.7/site-packages/antismash/common/secmet/record.py", line 764, in from_biopython
    raise SecmetInvalidInputError(str(err)) from err
antismash.common.secmet.errors.SecmetInvalidInputError: record does not contain domain referenced by module at [2331621:2334660](+): 'record NZ_LT629789.1 contains no domain named nrpspksdomains_ctg1_2108_Condensation_Starter.1'
```

```bash
pip freeze
antismash==5.1.1
bcbio-gff==0.6.6
biopython==1.76
cycler==0.10.0
helperlibs==0.1.11
Jinja2==2.11.1
joblib==0.14.1
kiwisolver==1.1.0
MarkupSafe==1.1.1
matplotlib==3.1.3
numpy==1.18.1
pyparsing==2.4.6
pyScss==1.3.5
pysvg-py3==0.2.2.post3
python-dateutil==2.8.1
scikit-learn==0.22.1
scipy==1.4.1
six==1.14.0
```

**Comment by SJShaw** [comment]
I came across this last week myself, it's caused by lines like this in the genbank file:
```
                     /domain_id="nrpspksdomains_ctg1_2108_Condensation_Starter.1
                     "
```
This very helpfully adds a trailing whitespace character which breaks the lookups, because I never expected there to be trailing whitespace added like that (or inserted, in some cases).

I've got a fix for it in #235 (specifically bfd0e19) which handles it when converting from genbank, and it should arrive in 5.1.2 or 5.2, whichever comes first. You can always include that patch in your local install in the meantime.

**Comment by seth-1** [comment]
Thanks @SJShaw!
